receiver/hostmetricsreceiver/receiver.go

```diff
@@ -0,0 +1,107 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package hostmetricsreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/sharedcomponent"
+	metadataPkg "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/experimentalmetricmetadata"
+)
+
+const entityType = "host"
+
+type hostMetricsReceiver struct {
+	cfg *Config
+
+	scraperHelper component.Component
+	nextLogs      consumer.Logs
+	cancel        context.CancelFunc
+
+	settings *receiver.Settings
+}
+
+func (hmr *hostMetricsReceiver) Start(ctx context.Context, host component.Host) error {
+	ctx, hmr.cancel = context.WithCancel(ctx)
+	if hmr.nextLogs != nil {
+		hmr.sendEntityEvent(ctx)
+		if hmr.cfg.MetadataCollectionInterval != 0 {
+			ticker := time.NewTicker(hmr.cfg.MetadataCollectionInterval)
+			go func() {
+				for {
+					select {
+					case <-ticker.C:
+						hmr.sendEntityEvent(ctx)
+					case <-ctx.Done():
+						ticker.Stop()
+						return
+					}
+				}
+			}()
+		}
+	}
+
+	if hmr.scraperHelper != nil {
+		return hmr.scraperHelper.Start(ctx, host)
+	}
+	return nil
+}
+
+func (hmr *hostMetricsReceiver) Shutdown(ctx context.Context) error {
+	if hmr.cancel != nil {
+		hmr.cancel()
+	}
+	if hmr.scraperHelper != nil {
+		return hmr.scraperHelper.Shutdown(ctx)
+	}
+	return nil
+}
+
+func newHostMetricsReceiver(cfg *Config, set *receiver.Settings) *hostMetricsReceiver {
+	r := &hostMetricsReceiver{
+		cfg:      cfg,
+		nextLogs: nil,
+		settings: set,
+	}
+
+	return r
+}
+
+func (hmr *hostMetricsReceiver) sendEntityEvent(ctx context.Context) {
+	timestamp := pcommon.NewTimestampFromTime(time.Now())
+
+	out := metadataPkg.NewEntityEventsSlice()
```

Comment:
In a follow up PR we should add "Interval" as a field to the Entity events and use `MetadataCollectionInterval` as the value. The "Interval" was proposed after the initial design and is now in [the OTEP](https://github.com/open-telemetry/oteps/blob/main/text/entities/0256-entities-data-model.md#entitystate-event).